homeassistant/components/notify/telstra.py

```diff
@@ -0,0 +1,90 @@
+"""
+Telstra API platform for notify component.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/notify.telstra/
+"""
+import logging
+
+import requests
+import voluptuous as vol
+
+from homeassistant.components.notify import (BaseNotificationService,
+                                             ATTR_TITLE,
+                                             PLATFORM_SCHEMA)
+import homeassistant.helpers.config_validation as cv
+
+CONF_CONSUMER_KEY = 'consumer_key'
+CONF_CONSUMER_SECRET = 'consumer_secret'
+CONF_PHONE_NUMBER = 'phone_number'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_CONSUMER_KEY): cv.string,
+    vol.Required(CONF_CONSUMER_SECRET): cv.string,
+    vol.Required(CONF_PHONE_NUMBER): cv.string,
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def get_service(hass, config):
+    """Get the Telstra API notification service."""
+    consumer_key = config.get(CONF_CONSUMER_KEY)
+    consumer_secret = config.get(CONF_CONSUMER_SECRET)
+    phone_number = config.get(CONF_PHONE_NUMBER)
+
```

Comment:
Please consider to check if the credentials are valid while you setup the service. This would ensure that the users do not end if a platform in their instance that is not functional.